### PR TITLE
Enable ha integration test

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following adds a generic enable-ha integration test, along with
removing ha to ensure that we are correctly going back from ha as 
well.

The code is actually simple and not that hard to follow, just waiting
for the machines to appear is most of the code.

## QA steps

*Please replace with how we can verify that the change works.*

```sh
(cd tests && ./main.sh controller)
```

### CODE REVIEW

File: tests/suites/controller/enable_ha.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
+wait_for_controller_machines() {
+    ammount=${1}
```
Comment (hmlanigan) [inline tests/suites/controller/enable_ha.sh:2]: ammount,  typo?

File: tests/suites/controller/enable_ha.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    attempt=0
+    until [[ "$(juju machines -m controller --format=json | jq -r ".machines | .[] | .[\"juju-status\"] | select(.current == \"started\") | .current" | wc -l | grep "${ammount}")" ]]; do
```
Comment (hmlanigan) [inline tests/suites/controller/enable_ha.sh:6]: This ensures that the machines have started.  Check for ha-status = ha-enabled on all 3 machines to ensure HA is up and running.

$ juju show-controller
  controller-machines:
    "0":
      instance-id: juju-a2cba4-0
      ha-status: ha-enabled
      ha-primary: true
    "1":
      instance-id: juju-a2cba4-1
      ha-status: ha-enabled
    "2":
      instance-id: juju-a2cba4-2
      ha-status: ha-enabled

Comment (hmlanigan) [reply tests/suites/controller/enable_ha.sh:6]: Or "controller-member-status": "has-vote" from juju machines.

**Review by hmlanigan — CHANGES_REQUESTED** [review]

File: tests/suites/controller/enable_ha.sh (hunk 2 above)
Comment (SimonRichardson) [reply tests/suites/controller/enable_ha.sh:6]: Do we care for this test? I know what you mean in the general case, but for this test we know we'll have exactly 3 machines for a controller...?

Comment (hmlanigan) [reply tests/suites/controller/enable_ha.sh:6]: We do, just because the machines deploy, doesn't mean that enable HA was successful.  The delays in the field have been after the machines were started.

**Review by hmlanigan — APPROVED** [review]
Thank you for the changes.  LGTM.  Eventually we should consider lowering the time allowed, or create a new test which has shorter timeframe.  To help catch items like lp: 184337

**Comment by SimonRichardson** [comment]
$$merge$$
